src/utils/core.ts

```diff
@@ -814,3 +814,11 @@ function removePrefix(token: string): string {
 	});
 	return token;
 }
+
+/**
+ * Detects whether a BOLT-11 Lightning invoice encodes a non-zero amount in the Human-Readable Part
+ * (HRP).
+ */
+export function invoiceHasAmountInHRP(invoice: string): boolean {
+	return /^ln[a-z]{2,}[1-9][0-9]*(?:[mun]|0p)?1/i.test(invoice);
```

Comment:
The regex doesn't handle `p` (pico) invoices correctly, is that intentional?  or should it be `(?:[munp])?` the 0 seems to always be consumed as part of the amount.

Reply:
The pico handling is intended as bolt11 spec requires:

If the p multiplier is used the last decimal of amount MUST be 0.

https://github.com/lightning/bolts/blob/master/11-payment-encoding.md#requirements-1

Reply:
No worries, yeah, they are a bit (pun intended).

But a vaild point - we should add a pico invoice to the tests.